.github/workflows/action.yml

```diff
@@ -48,46 +49,83 @@ jobs:
-      run: az extension add --name spring-cloud
-   
-    - name: Deploy with AZ CLI commands
-      run: |
-          az configure --defaults group=$RESOURCE_GROUP
-          az configure --defaults spring-cloud=$SPRING_CLOUD_SERVICE
-          az spring-cloud app deploy --name $API_GATEWAY \
-            --jar-path ${{ github.workspace }}/$API_GATEWAY_JAR \
-            --jvm-options='-Xms2048m -Xmx2048m -Dspring.profiles.active=mysql'
-          az spring-cloud app deploy --name $ADMIN_SERVER \
-            --jar-path ${{ github.workspace }}/$ADMIN_SERVER_JAR \
-            --jvm-options='-Xms2048m -Xmx2048m -Dspring.profiles.active=mysql'
-          az spring-cloud app deploy --name $CUSTOMERS_SERVICE \
-            --jar-path ${{ github.workspace }}/$CUSTOMERS_SERVICE_JAR \
-            --jvm-options='-Xms2048m -Xmx2048m -Dspring.profiles.active=mysql' \
-            --env MYSQL_SERVER_FULL_NAME=${{ steps.mysqlsecrets.outputs.MYSQL-SERVER-FULL-NAME }} \
-              MYSQL_DATABASE_NAME=${{ steps.mysqlsecrets.outputs.MYSQL-DATABASE-NAME }} \
-              MYSQL_SERVER_ADMIN_LOGIN_NAME=${{ steps.mysqlsecrets.outputs.MYSQL-SERVER-ADMIN-LOGIN-NAME }} \
-              MYSQL_SERVER_ADMIN_PASSWORD=${{ steps.mysqlsecrets.outputs.MYSQL-SERVER-ADMIN-PASSWORD }}
-          az spring-cloud app deploy --name $VETS_SERVICE \
-            --jar-path ${{ github.workspace }}/$VETS_SERVICE_JAR \
-            --jvm-options='-Xms2048m -Xmx2048m -Dspring.profiles.active=mysql' \
-            --env MYSQL_SERVER_FULL_NAME=${{ steps.mysqlsecrets.outputs.MYSQL-SERVER-FULL-NAME }} \
-              MYSQL_DATABASE_NAME=${{ steps.mysqlsecrets.outputs.MYSQL-DATABASE-NAME }} \
-              MYSQL_SERVER_ADMIN_LOGIN_NAME=${{ steps.mysqlsecrets.outputs.MYSQL-SERVER-ADMIN-LOGIN-NAME }} \
-              MYSQL_SERVER_ADMIN_PASSWORD=${{ steps.mysqlsecrets.outputs.MYSQL-SERVER-ADMIN-PASSWORD }}
-          az spring-cloud app deploy --name $VISITS_SERVICE \
-            --jar-path ${{ github.workspace }}/$VISITS_SERVICE_JAR \
-            --jvm-options='-Xms2048m -Xmx2048m -Dspring.profiles.active=mysql' \
-            --env MYSQL_SERVER_FULL_NAME=${{ steps.mysqlsecrets.outputs.MYSQL-SERVER-FULL-NAME }} \
-              MYSQL_DATABASE_NAME=${{ steps.mysqlsecrets.outputs.MYSQL-DATABASE-NAME }} \
-              MYSQL_SERVER_ADMIN_LOGIN_NAME=${{ steps.mysqlsecrets.outputs.MYSQL-SERVER-ADMIN-LOGIN-NAME }} \
-              MYSQL_SERVER_ADMIN_PASSWORD=${{ steps.mysqlsecrets.outputs.MYSQL-SERVER-ADMIN-PASSWORD }}
+      
+    - name: Deploy api-gateway
+      uses: Azure/spring-cloud-deploy@v1
+      with:
+        azure-subscription: ${{ env.AZURE_SUBSCRIPTION }}
+        action: deploy
+        service-name: ${{ env.SPRING_CLOUD_SERVICE }}
+        app-name: ${{ env.API_GATEWAY }}
+        use-staging-deployment: false
+        package: ${{ github.workspace }}/${{ env.API_GATEWAY_JAR }}
+        # jvm-options: -Xms2048m -Xmx2048m -Dspring.profiles.active=mysql
```

Comment:
jvm options not working yet due to a bug, will add these lines back before merge.